### PR TITLE
autotest: introduce pytest-env plugin

### PR DESCRIPTION
**Pull request by miurahr**
## What does this PR do?

pytest-env plugin enables feature to set environment
variable in pytest.ini configuration file.
We sometimes necessary to set GDAL_DATA and other
test configuration variables for customized environment,
and the plugin help it.

Signed-off-by: Hiroshi Miura <miurahr@linux.com>

## What are related issues/pull requests?

N.A.

## Tasklist

 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

### CODE REVIEW

File: autotest/pytest.ini [hunk 1]
```diff
@@ -1,6 +1,14 @@
...
 python_files = *.py
 testpaths = ogr gcore gdrivers osr alg gnm utilities pyscripts
+log_file = autotest.log
+log_file_level = INFO
+env =
+  GDAL_DOWNLOAD_TEST_DATA=NO
+  GDAL_RUN_SLOW_TESTS=NO
+  LD_LIBRARY_PATH=../gdal
+  GDAL_DATA=../gdal/data
+#  GDAL_DRIVER_PATH=/usr/lib/gdal/gdalplugins
```
Comment (miurahr) [inline autotest/pytest.ini]: There would be a discussion what is default here or just add it for help as commented out.

**Comment by rouault** [comment]
Is it needed ? There is the scripts/setdevenv.sh that does all the necessary env variable setting.
CC @craigds

**Comment by miurahr** [comment]
With this extension, developer can run pytest with environment values without using shell script and power shell wrapper.

File: autotest/pytest.ini (hunk 1 above)
Comment (miurahr) [reply autotest/pytest.ini]: These are examples so commented out.

**Pull request merged**